imagepipeline-base/src/main/java/com/facebook/imageformat/ImageFormatCheckerUtils.java

```diff
@@ -59,5 +59,35 @@ public static boolean startsWithPattern(
     return true;
   }
 
+  /**
+   * Checks if byteArray interpreted as sequence of bytes contains the pattern.
+   * @param byteArray the byte array to be checked
+   * @param pattern the pattern to check
+   * @return index of beginning of pattern, if found; otherwise -1
+   */
+  public static int indexOfPattern(
+      final byte[] byteArray,
+      final byte[] pattern) {
+    Preconditions.checkNotNull(byteArray);
+    Preconditions.checkNotNull(pattern);
+    if (pattern.length > byteArray.length) {
+      return -1;
+    }
+
+    for (int i = 0; i < (byteArray.length - pattern.length); i++) {
```

Comment:
I'm not sure either, but I fixed this when I remodeled the whole method to look more like the String.indexOf() method you referenced in your comment below.